packages/pages/src/dev/server/ssr/generateTestData.ts

```diff
@@ -294,3 +298,62 @@ const getDocumentByLocale = (parsedData: any, locale: string): any => {
   }
   return parsedData;
 };
+
+const getDocumentBySlug = (
+  parsedData: any,
+  slug: string,
+  slugFields: string[],
+  templateModuleCollection: TemplateModuleCollection
+): any => {
+  if (!Array.isArray(parsedData)) {
+    return parsedData;
+  }
+
+  // Find the slugField where the slug value matches
+  const matchingSlugFieldsSet: Set<string> = new Set();
+  for (const document of parsedData) {
+    for (const slugField of slugFields) {
+      if (document[slugField] === slug) {
+        matchingSlugFieldsSet.add(slugField);
+      }
+    }
+  }
+
+  const matchingSlugFields = Array.from(matchingSlugFieldsSet);
+
+  if (matchingSlugFields.length === 0) {
+    throw new Error(`Could not find slugField on document for slug: "${slug}"`);
+  } else if (matchingSlugFields.length > 1) {
+    throw new Error(
+      `Multiple documents have the same slugField (${matchingSlugFields.join(
+        ", "
+      )}) with value: "${slug}"`
+    );
+  }
+
+  // Find the template that uses the slugfield
+  const matchingTemplateModules: TemplateModuleInternal<any, any>[] = [];
+  templateModuleCollection.forEach((templateModule) => {
+    const slugField = templateModule.config?.slugField || "slug";
```

Comment:
random but do we prefer ?? over ||

Reply:
`??` does nullish coalescing, so empty string will be taken as truthy and returned (if it happens to be empty string). We don't want that.

https://stackoverflow.com/a/61481631